src/lib/gadgets/gadgets.ts

```diff
@@ -33,4 +34,32 @@ const Gadgets = {
   rangeCheck64(x: Field) {
     return rangeCheck64(x);
   },
+
+  /**
+   * A (left and right) rotation is similar to the shift operation, `<<` and `>>` in JavaScript, just that bits are being appended to the other side.
+   * `direction` is a string which accepts either `'left'` or `'right'`, defining the direction of the rotation.
+   *
+   * **Note:** You can not rotate {@link Field} elements that exceed 64 bits. For elements that exceed 64 bits this operation will fail.
+   *
+   * @param word {@link Field} element to rotate.
+   * @param bits amount of bits to rotate this {@link Field} element with.
+   * @param direction left or right rotation direction.
+   *
+   * @throws Throws an error if the input value exceeds 64 bits.
+   *
+   * @example
+   * ```ts
+   * const x = Provable.witness(Field, () => Field(12));
+   * const y = rot(x, 2, 'left'); // left rotation by 2 bits
+   * const z = rot(x, 2, 'right'); // right rotation by 2 bits
+   * y.assertEquals(48);
+   * z.assertEquals(3)
+   *
+   * const xLarge = Provable.witness(Field, () => Field(12345678901234567890123456789012345678n));
+   * rot(xLarge, 32, "left"); // throws an error since input exceeds 64 bits
+   * ```
+   */
+  rot(word: Field, bits: number, direction: 'left' | 'right' = 'left') {
```

Comment:
One thought we could do here instead is instead of taking `word` as a `Field`, we specify its type to only be `UInt64`. Any thoughts?

Reply:
My preference would be to keep the gadgets "pure" and only accept field elements, one reason for that is gadgets and gates naturally feel low level to me, so I think it would be a good idea to only feed them with field elements. Another reason is that algorithms built on top of it might operate on field element by default (e.g. because its better DX or other gates that aren't constrained by bit size do) - `UInt64 <> Field` conversions might become a bit tedious in the future

Reply:
Sounds like great reasoning!